Makefile

```diff
@@ -26,70 +26,113 @@ MAKEFLAGS += -j$(shell nproc)
+# But with make 4.3 that changes the entire Makefile to be serial.
+
+# Here we make dynamic prereqs to apply to other targets that need to run in sequence.
+FORMAT_PREREQS :=
+
+.PHONY: format
+format: build/format.${CONDA_ENV_NAME}.build-stamp
+
+ifneq (,$(filter format,$(MAKECMDGOALS)))
+    FORMAT_PREREQS += build/format.${CONDA_ENV_NAME}.build-stamp
+endif
+
+build/format.${CONDA_ENV_NAME}.build-stamp: build/licenseheaders.${CONDA_ENV_NAME}.build-stamp
+# TODO: add isort and black formatters
+build/format.${CONDA_ENV_NAME}.build-stamp:
+	touch $@
+
+.PHONY: licenseheaders
+licenseheaders: build/licenseheaders.${CONDA_ENV_NAME}.build-stamp
+
+ifneq (,$(filter licenseheaders,$(MAKECMDGOALS)))
+    FORMAT_PREREQS += build/licenseheaders.${CONDA_ENV_NAME}.build-stamp
+endif
+
+build/licenseheaders.${CONDA_ENV_NAME}.build-stamp: build/conda-env.${CONDA_ENV_NAME}.build-stamp
+build/licenseheaders.${CONDA_ENV_NAME}.build-stamp: $(PYTHON_FILES)
+build/licenseheaders.${CONDA_ENV_NAME}.build-stamp: $(SCRIPT_FILES)
+build/licenseheaders.${CONDA_ENV_NAME}.build-stamp: $(SQL_FILES) doc/mit-license.tmpl
+build/licenseheaders.${CONDA_ENV_NAME}.build-stamp: doc/mit-license.tmpl
+build/licenseheaders.${CONDA_ENV_NAME}.build-stamp:
+	# Note: to avoid makefile dependency loops, we don't touch the setup.py
+	# files as that would force the conda-env to be rebuilt.
+	conda run -n ${CONDA_ENV_NAME} licenseheaders -t doc/mit-license.tmpl \
+		-E .py .sh .ps1 .sql .cmd \
+		-x mlos_bench/setup.py mlos_core/setup.py mlos_viz/setup.py
+	touch $@
+
+
 .PHONY: check
-check: pycodestyle pydocstyle pylint mypy # cspell licenseheaders markdown-link-check
+check: pycodestyle pydocstyle pylint mypy # cspell markdown-link-check
+# TODO: add isort and black checks
 
 .PHONY: pycodestyle
-pycodestyle: conda-env
 pycodestyle: build/pycodestyle.mlos_core.${CONDA_ENV_NAME}.build-stamp
 pycodestyle: build/pycodestyle.mlos_bench.${CONDA_ENV_NAME}.build-stamp
 pycodestyle: build/pycodestyle.mlos_viz.${CONDA_ENV_NAME}.build-stamp
 
-
 build/pycodestyle.mlos_core.${CONDA_ENV_NAME}.build-stamp: $(MLOS_CORE_PYTHON_FILES)
 build/pycodestyle.mlos_bench.${CONDA_ENV_NAME}.build-stamp: $(MLOS_BENCH_PYTHON_FILES)
 build/pycodestyle.mlos_viz.${CONDA_ENV_NAME}.build-stamp: $(MLOS_VIZ_PYTHON_FILES)
 
-build/pycodestyle.%.${CONDA_ENV_NAME}.build-stamp: build/conda-env.${CONDA_ENV_NAME}.build-stamp setup.cfg
+PYCODESTYLE_COMMON_PREREQS := build/conda-env.${CONDA_ENV_NAME}.build-stamp
+PYCODESTYLE_COMMON_PREREQS += $(FORMAT_PREREQS)
+PYCODESTYLE_COMMON_PREREQS += $(MLOS_GLOBAL_CONF_FILES)
+
+build/pycodestyle.%.${CONDA_ENV_NAME}.build-stamp: $(PYCODESTYLE_COMMON_PREREQS)
 	# Check for decent pep8 code style with pycodestyle.
-	# Note: if this fails, try using autopep8 to fix it.
-	conda run -n ${CONDA_ENV_NAME} pycodestyle $(filter-out setup.cfg,$+)
+	# Note: if this fails, try using 'make format' to fix it.
+	conda run -n ${CONDA_ENV_NAME} pycodestyle $(filter %.py,$+)
```

Comment:
Only run on `.py` files and declare using `make format` to fix instead of some tool in particular.

Eventually that allows us to replace it with `black`